pkg/controller/drain/drain_controller.go

```diff
@@ -303,12 +312,49 @@ func (ctrl *Controller) syncNode(key string) error {
 	desiredVerb := strings.Split(desiredState, "-")[0]
 	switch desiredVerb {
 	case daemonconsts.DrainerStateUncordon:
+		err = upgrademonitor.GenerateAndApplyMachineConfigNodes(&upgrademonitor.Condition{State: v1alpha1.MachineConfigNodeUpdateComplete, Reason: string(v1alpha1.MachineConfigNodeUpdateCordoned), Message: fmt.Sprintf("(Un)Cordoning Node as part of completing upgrade phase")},
+			&upgrademonitor.Condition{State: v1alpha1.MachineConfigNodeUpdateCordoned, Reason: fmt.Sprintf("%s%s", string(v1alpha1.MachineConfigNodeUpdateComplete), string(v1alpha1.MachineConfigNodeUpdateCordoned)), Message: fmt.Sprintf("(Un)Cordoning node. The node is reporting Unschedulable = %t", node.Spec.Unschedulable)},
```

Comment:
(Un)cordons should generally be almost instant, or fail. Since we have the failure covered, WDYT about merging this and the below (Un)cordon'ed (so we only report afterwards)

Reply:
sure. I think since it is quick no one will care if we just skip to true.